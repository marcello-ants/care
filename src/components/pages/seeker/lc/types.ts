import { GeneralProviderProfile } from '@/types/seeker';

export interface ProviderProfile extends GeneralProviderProfile {
  numberOfReviews: number;
  qualities?: ProviderQualities;
  services?: ProviderServices;
  firstName: string;
}

export interface ProviderQualities {
  certifiedNursingAssistant?: boolean;
  certifiedRegisterNurse?: boolean;
  certifiedTeacher?: boolean;
  childDevelopmentAssociate?: boolean;
  comfortableWithPets?: boolean;
  cprTrained?: boolean;
  crn?: boolean;
  doesNotSmoke?: boolean;
  doula?: boolean;
  earlyChildDevelopmentCoursework?: boolean;
  earlyChildhoodEducation?: boolean;
  firstAidTraining?: boolean;
  nafccCertified?: boolean;
  ownTransportation?: boolean;
  specialNeedsCare?: boolean;
  trustlineCertifiedCalifornia?: boolean;
}

export interface ProviderServices {
  carpooling?: boolean;
  craftAssistance?: boolean;
  errands?: boolean;
  groceryShopping?: boolean;
  laundryAssistance?: boolean;
  lightHousekeeping?: boolean;
  mealPreparation?: boolean;
  swimmingSupervision?: boolean;
  travel?: boolean;
}
