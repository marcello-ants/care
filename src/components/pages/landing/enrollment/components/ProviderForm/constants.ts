export const PROVIDER_SERVICES_DESKTOP = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const PROVIDER_SERVICES_MOBILE = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const DROPOFF_URLS = {
  CHILD_CARE: '/swf/ccOptimizedFlow',
  SENIOR_CARE: '/app/enrollment/provider/sc ',
  PET_CARE: '/swf/pcOptimizedFlow',
  HOUSEKEEPING: '/swf/hkOptimizedFlow',
  TUTORING: '/swf/tuOptimizedFlow',
  SPECIAL_NEEDS: '/swf/snOptimizedFlow',
  GENERIC: '/swf/cgOptimizedFlow',
  AU_PAIR: '',
  CARE_GIGS: '',
};
