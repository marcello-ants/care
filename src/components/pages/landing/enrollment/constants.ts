export const SITTER_FORM = {
  providerServices: {
    title: 'What Services do you offer?',
    id: 'providerService',
    options: [
      {
        value: 'CHILD_CARE',
        text: 'Babysitter',
      },
      {
        value: 'CHILD_CARE',
        text: 'Nanny',
      },
      {
        value: 'SPECIAL_NEEDS',
        text: 'Special Needs Provider',
      },
      {
        value: 'TUTORING',
        text: 'Tutor',
      },
      {
        value: 'TUTORING',
        text: 'Private Lesson Instructor',
      },
      {
        value: 'SENIOR_CARE',
        text: 'Senior Care Provider',
      },
      {
        value: 'PET_CARE',
        text: 'Pet Care Provider',
      },
      {
        value: 'HOUSEKEEPING',
        text: 'Housekeeper',
      },
      {
        value: 'GENERIC',
        text: 'Errands &amp; Odd Jobs Provider',
      },
    ],
  },
  providerEmail: {
    id: 'email',
    placeholder: 'Email',
  },
  providerPassword: {
    id: 'password',
    placeholder: 'Password',
  },
  providerZip: {
    id: 'providerZip',
    placeholder: 'ZIP',
  },
};

export const GENERIC_FORM = {
  howDidYouHearAboutUs: {
    title: 'How did you hear about us?',
    id: 'howDidYouHearAboutUs',
    options: [
      {
        value: 'FACEBOOK',
        text: 'Facebook',
      },
      {
        value: 'BANNER_AD',
        text: 'Banner Ad',
      },
      {
        value: 'TV_AD',
        text: 'TV Ad',
      },
      {
        value: 'OTHER_SOCIAL_MEDIA',
        text: 'Other Social Media (Twitter, Instagram, Pinterest, LinkedIn)',
      },
      {
        value: 'PRESS_COVERAGE',
        text: 'Press Coverage (News, Magazine, Blog)',
      },
      {
        value: 'FRIENDS_OR_FAMILY',
        text: 'Friends or Family',
      },
      {
        value: 'ONLINE_VIDEO',
        text: 'Online Video (Hulu, Youtube)',
      },
      {
        value: 'SEARCH_ENGINE',
        text: 'Search Engine (Google, Bing)',
      },
      {
        value: 'PARENTING_GROUP',
        text: 'Parenting Group or Forum',
      },
      {
        value: 'OTHER',
        text: 'Other',
      },
    ],
  },
};
