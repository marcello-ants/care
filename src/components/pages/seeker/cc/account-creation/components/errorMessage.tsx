import React from 'react';
import { Box, Grid } from '@material-ui/core';
import { Alert } from '@material-ui/lab';

type ErrorMessageData = {
  errorMessageText: string;
};

const ErrorMessageNotification = (props: ErrorMessageData) => {
  const { errorMessageText } = props;

  return (
    <Grid item xs={12}>
      <Box mb={3}>
        <Alert severity="error">{errorMessageText}</Alert>
      </Box>
    </Grid>
  );
};

export default ErrorMessageNotification;
