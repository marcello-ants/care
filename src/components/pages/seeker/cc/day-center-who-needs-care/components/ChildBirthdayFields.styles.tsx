// External Dependencies
import { makeStyles } from '@material-ui/core';
import { theme } from '@care/material-ui-theme';

const useStyles = makeStyles({
  childContainer: {
    padding: theme.spacing(0),
  },
  childContainerNth: {
    marginTop: theme.spacing(4),
    position: 'relative',
  },
  optionsContainer: {
    display: 'flex',
    flexDirection: 'row',
  },
  fullWidth: {
    width: '100%',
  },
  error: {
    color: theme.palette.care.red[800],
    paddingLeft: theme.spacing(1),
  },
  selectContainer: {
    marginTop: theme.spacing(1),
  },
  monthSelect: {
    marginRight: theme.spacing(0.5),
    '& label': {
      paddingLeft: theme.spacing(1),
    },
  },
  yearSelect: {
    marginLeft: theme.spacing(0.5),
    '& label': {
      paddingLeft: theme.spacing(1),
    },
  },
  removeChildButton: {
    position: 'absolute',
    top: '0',
    right: '0',
    width: 'auto',
    height: 'auto',
    lineHeight: 'unset',
    minWidth: 'auto',
    padding: theme.spacing(0),
    color: theme.palette.care.blue[700],
    '&:hover': {
      backgroundColor: 'transparent',
    },
  },
});

export default useStyles;
