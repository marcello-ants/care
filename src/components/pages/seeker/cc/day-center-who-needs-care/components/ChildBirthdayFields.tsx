// External Dependencies
import React, { useMemo } from 'react';
import { Button, FormHelperText, Grid, InputLabel, MenuItem } from '@material-ui/core';
import { Select, Typography } from '@care/react-component-lib';

// Internal Dependencies
import { ChildDateOfBirth } from '@/types/seekerCC';
import { getBirthAvailableMonthsAndYears } from '../utils';
import useStyles from './ChildBirthdayFields.styles';

// Interfaces
interface TErrorProps {
  year?: string;
  month?: string;
}

type SelectInputProps = React.ComponentProps<typeof Select>;
interface IChildBirthdayFieldsProps {
  child: ChildDateOfBirth;
  onRemoveChild: () => void;
  onChange: SelectInputProps['onChange'];
  childIndex: number;
  error?: TErrorProps;
  allow9MonthsOutFromNow?: boolean;
}

function ChildBirthdayFields({
  child,
  childIndex: i,
  onChange,
  onRemoveChild,
  error,
  allow9MonthsOutFromNow,
}: IChildBirthdayFieldsProps) {
  const classes = useStyles();

  const { months, years } = useMemo(
    () => getBirthAvailableMonthsAndYears(child.month, child.year, allow9MonthsOutFromNow),
    [child.year, child.month, allow9MonthsOutFromNow]
  );

  const monthsOptions = useMemo(() => {
    return months.map((month) => {
      return (
        <MenuItem key={month.value} value={month.value}>
          {month.name}
        </MenuItem>
      );
    });
  }, [months]);

  const yearsOptions = useMemo(() => {
    return years.map((year: string) => {
      return (
        <MenuItem key={year} value={year}>
          {year}
        </MenuItem>
      );
    });
  }, [years]);

  return (
    <Grid
      // eslint-disable-next-line react/no-array-index-key
      item
      xs={12}
      className={`${
        i === 0
          ? `${classes.childContainer}`
          : `${classes.childContainer} ${classes.childContainerNth}`
      }`}>
      <Typography variant="h3">Child {i + 1}</Typography>
      {i !== 0 && (
        <Button className={classes.removeChildButton} onClick={onRemoveChild}>
          Remove Child
        </Button>
      )}
      <Grid item xs={12} className={classes.optionsContainer}>
        <Grid item xs={6} className={`${classes.selectContainer} ${classes.monthSelect}`}>
          <InputLabel id={`month-select-label-${i}`}>Birth month</InputLabel>
          <Select
            className={classes.fullWidth}
            labelId={`month-select-label-${i}`}
            id={`month-select-child-${i}`}
            name={`children[${i}].month`}
            onChange={onChange}
            value={child.month}
            error={Boolean(error?.month)}>
            {monthsOptions}
          </Select>
          {error?.month && (
            <FormHelperText className={classes.error}>Select their birth month</FormHelperText>
          )}
        </Grid>
        <Grid item xs={6} className={`${classes.selectContainer} ${classes.yearSelect}`}>
          <InputLabel id={`year-select-label-${i}`}>Birth year</InputLabel>
          <Select
            className={classes.fullWidth}
            labelId={`year-select-label-${i}`}
            id={`year-select-child-${i}`}
            name={`children[${i}].year`}
            onChange={onChange}
            value={child.year}
            error={Boolean(error?.year)}>
            {yearsOptions}
          </Select>
          {error?.year && (
            <FormHelperText className={classes.error}>Select their birth year</FormHelperText>
          )}
        </Grid>
      </Grid>
    </Grid>
  );
}

ChildBirthdayFields.defaultProps = {
  error: undefined,
  allow9MonthsOutFromNow: undefined,
};

export type { IChildBirthdayFieldsProps, TErrorProps };
export default ChildBirthdayFields;
