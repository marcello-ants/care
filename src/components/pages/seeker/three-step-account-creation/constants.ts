import * as yup from 'yup';
import { ServiceType } from '@/__generated__/globalTypes';
import { VerticalsAbbreviation } from '@/constants';

export const validationSchema = yup.object({
  firstName: yup
    .string()
    .min(2, 'First name needs to have at least two letters.')
    .required('First name is required'),
  lastName: yup
    .string()
    .min(2, 'Last name needs to have at least two letters.')
    .required('Last name is required'),
});

export const SERVICE_TYPE_BY_VERTICAL: { [key: string]: ServiceType } = {
  HK: ServiceType.HOUSEKEEPING,
  PC: ServiceType.PET_CARE,
  TU: ServiceType.TUTORING,
  CC: ServiceType.CHILD_CARE,
  DC: ServiceType.CHILD_CARE,
  IB: ServiceType.CHILD_CARE,
  SC: ServiceType.SENIOR_CARE,
};

export const ERROR_CREATING_ACCOUNT = 'An error occurred creating your account, please try again.';

export const FACEBOOK_VERTICALS_ALLOWANCE = ['CC'];
export const HDYHAU_VERTICALS_ALLOWANCE: VerticalsAbbreviation[] = ['HK', 'TU', 'PC', 'DC'];
