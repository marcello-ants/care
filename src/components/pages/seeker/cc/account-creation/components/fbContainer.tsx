import { Box, makeStyles, Grid, Button } from '@material-ui/core';
import { useFacebookApiKey } from '@/components/FacebookApiKeyContext';

// @ts-ignore
import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props';
import { Icon48SocialMediaFacebook } from '@care/react-icons';

export type facebookResponseProps = {
  accessToken: string;
  // eslint-disable-next-line
  data_access_expiration_time?: number;
  email: string;
  expiresIn?: number;
  graphDomain?: string;
  id?: string;
  name: string;
  signedRequest?: string;
  userID?: string;
};

export interface SplitFormValues {
  email: string;
}

export interface SplitFormWithFbData extends SplitFormValues {
  firstName?: string;
  lastName?: string;
  accessToken?: string;
}

const mobileWidth = 360;
const fbColor = '#1877F2';
const fbIconSize = '35px';
const useStyles = (fbEmphasizedDesign: boolean) =>
  makeStyles((theme) => ({
    fbButton: {
      paddingTop: theme.spacing(5),
      borderColor: fbEmphasizedDesign ? fbColor : theme.palette.default?.dark,
      background: fbEmphasizedDesign ? fbColor : theme.palette.care?.white,
      color: fbEmphasizedDesign ? theme.palette.care?.white : theme.palette.default?.dark,
      fontSize: fbEmphasizedDesign ? '21px' : '18px',
      '&:hover': {
        background: fbEmphasizedDesign ? fbColor : theme.palette.care?.white,
      },
    },
    fbButtonContent: {
      display: 'grid',
      gridTemplateColumns: 'repeat(3, minmax(0, 1fr))',
      [theme.breakpoints.down(mobileWidth)]: {
        display: 'flex',
        alignItems: 'center',
      },
    },
    fbIconContainer: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: fbIconSize,
      '& svg': {
        width: fbIconSize,
        height: fbIconSize,
      },
    },
    fbIcon: {
      background: '#1a77f2',
      borderRadius: '19px',
    },
    fbButtonText: {
      justifySelf: 'center',
      marginLeft: '10px',
      marginRight: '5px',
      fontColor: theme.palette.care?.white,
    },
  }));

type GetFbContainerProps = {
  handleFacebookLogin: (response: facebookResponseProps) => Promise<void>;
  fbButtonOnTop: boolean;
  fbButtonOnBottom: boolean;
};

const getFbContainer = ({
  handleFacebookLogin,
  fbButtonOnTop,
  fbButtonOnBottom,
}: GetFbContainerProps) => {
  const fbEmphasizedDesign = fbButtonOnBottom || fbButtonOnTop;
  const classes = useStyles(fbEmphasizedDesign)();
  const { apiKey } = useFacebookApiKey();

  return (
    <Grid item xs={12}>
      <Box pt={fbButtonOnTop ? 0 : 2}>
        <FacebookLogin
          appId={apiKey}
          autoLoad={false}
          fields="name,email"
          callback={handleFacebookLogin}
          render={(renderProps: any) => (
            <Button
              onClick={renderProps.onClick}
              className={classes.fbButton}
              variant="outlined"
              size="large"
              fullWidth>
              <div className={classes.fbButtonContent}>
                <div className={classes.fbIconContainer}>
                  <Icon48SocialMediaFacebook size="38px" color="white" className={classes.fbIcon} />
                </div>
                <p className={classes.fbButtonText}>Continue with Facebook</p>
              </div>
            </Button>
          )}
        />
      </Box>
    </Grid>
  );
};

export default getFbContainer;
