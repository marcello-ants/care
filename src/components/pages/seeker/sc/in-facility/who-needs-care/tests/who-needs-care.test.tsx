import { NextRouter } from 'next/router';
import { screen, fireEvent, within, waitFor } from '@testing-library/react';
import { MockedResponse } from '@apollo/client/testing';

import { FeatureFlags } from '@/components/FeatureFlagsContext';
import { GET_SEEKER_INFO, GET_SEEKER_ZIP_CODE } from '@/components/request/GQL';
import * as AppStateComponent from '@/components/AppState';

import { initialAppState } from '@/state';
import { CLIENT_FEATURE_FLAGS, SEEKER_IN_FACILITY_ROUTES } from '@/constants';
import { preRenderPage } from '@/__setup__/testUtil';
import { AppState } from '@/types/app';
import { AnalyticsHelper } from '@/utilities/analyticsHelper';

import WhoNeedsInfacilityCare from '../who-needs-care';

const getSeekerInfoMock = {
  request: {
    operationName: 'getSeekerInfo',
    variables: { memberId: '123' },
    query: GET_SEEKER_INFO,
  },
  result: {
    data: {
      getSeeker: {
        member: {
          firstName: 'John',
          lastName: 'Doe',
          contact: {
            primaryPhone: '',
            __typename: 'MemberContact',
          },
          email: 'sc-1769-3@test.com',
          __typename: 'Member',
        },
        __typename: 'Seeker',
      },
    },
  },
};

describe('Who needs care', () => {
  describe('In facility', () => {
    let mockRouter: NextRouter | null;
    let asFragment: () => DocumentFragment;

    const renderComponent = (
      initialState: AppState,
      ldFlags: FeatureFlags = {},
      pathName: string = '/seeker/sc/help-type',
      mocks: MockedResponse[] = []
    ) => {
      const utils = preRenderPage({
        appState: initialState,
        flags: ldFlags,
        pathname: pathName,
        mocks,
      });
      mockRouter = utils.routerMock;
      ({ asFragment } = utils.renderFn(WhoNeedsInfacilityCare));
    };

    afterEach(() => {
      // cleanup on exiting
      mockRouter = null;
    });

    it('matches snapshot', () => {
      renderComponent(initialAppState);
      expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly', () => {
      renderComponent(initialAppState);
      expect(
        screen.getByText('Let’s get started by learning more about your senior care needs')
      ).toBeInTheDocument();
      expect(screen.getByText('Who needs care?')).toBeInTheDocument();
      expect(screen.getByText('How old are they?')).toBeInTheDocument();
      // makes sure it's defaulted to 80's
      expect(screen.getByText(/80's/)).toBeInTheDocument();

      expect(screen.getByRole('button', { name: 'Next' })).toBeDisabled();
    });

    it('should enable next button when selecting an option', async () => {
      renderComponent(initialAppState);
      expect(screen.getByText('Who needs care?')).toBeInTheDocument();
      fireEvent.click(screen.getByText('Other'));

      fireEvent.mouseDown(screen.getByText(/80's/));
      const listbox = within(screen.getByRole('listbox'));
      fireEvent.click(listbox.getByText("30's"));

      await waitFor(() => expect(screen.getByRole('button', { name: 'Next' })).toBeEnabled());

      fireEvent.click(screen.getByRole('button', { name: 'Next' }));

      expect(mockRouter!.push).toHaveBeenCalledWith(SEEKER_IN_FACILITY_ROUTES.DESCRIBE_LOVED_ONE);
    });

    it('Should route to /location when ENROLLMENT_RECOMMENDATIONS_FLOW_OPTIMIZATION is on', () => {
      jest.spyOn(AnalyticsHelper, 'logTestExposure');

      const flags: FeatureFlags = {
        [CLIENT_FEATURE_FLAGS.ENROLLMENT_RECOMMENDATIONS_FLOW_OPTIMIZATION]: {
          reason: {
            kind: 'OVERRIDE',
          },
          value: '',
          variationIndex: 1,
        },
      };
      const { renderFn, routerMock } = preRenderPage({ flags });
      renderFn(WhoNeedsInfacilityCare);

      expect(
        screen.queryByText(/Let’s get started by learning more about your senior care needs/)
      ).not.toBeInTheDocument();
      expect(AnalyticsHelper.logTestExposure).toHaveBeenCalledWith(
        CLIENT_FEATURE_FLAGS.ENROLLMENT_RECOMMENDATIONS_FLOW_OPTIMIZATION,
        flags[CLIENT_FEATURE_FLAGS.ENROLLMENT_RECOMMENDATIONS_FLOW_OPTIMIZATION]
      );
      expect(routerMock.push).toHaveBeenCalledWith(SEEKER_IN_FACILITY_ROUTES.LOCATION);
    });

    it('Should prevent screen view event to be fired when ENROLLMENT_RECOMMENDATIONS_FLOW_OPTIMIZATION is on', () => {
      jest.spyOn(AnalyticsHelper, 'logEvent');

      const flags: FeatureFlags = {
        [CLIENT_FEATURE_FLAGS.ENROLLMENT_RECOMMENDATIONS_FLOW_OPTIMIZATION]: {
          reason: {
            kind: 'OVERRIDE',
          },
          value: '',
          variationIndex: 1,
        },
      };
      const { renderFn } = preRenderPage({ flags });
      renderFn(WhoNeedsInfacilityCare);
      expect(AnalyticsHelper.logEvent).not.toHaveBeenCalledWith({
        name: 'Screen Viewed',
        data: {
          lead_flow: 'mhp module',
        },
      });
      expect(AnalyticsHelper.logEvent).not.toHaveBeenCalledWith({
        name: 'Screen Viewed',
        data: {},
      });
    });

    it('should fire screen view event', async () => {
      jest.spyOn(AnalyticsHelper, 'logEvent');
      const { renderFn } = preRenderPage();

      renderFn(WhoNeedsInfacilityCare);
      expect(AnalyticsHelper.logEvent).toHaveBeenCalledWith({
        name: 'Screen Viewed',
        data: {},
      });
    });
  });

  const getZipCodeMock = {
    request: {
      query: GET_SEEKER_ZIP_CODE,
      variables: {
        memberId: '123',
      },
    },
    result: {
      data: {
        getSeeker: {
          member: {
            address: {
              zip: '90001',
            },
          },
        },
      },
    },
  };

  describe('In facility nth day', () => {
    const nthDayState: AppState = {
      ...initialAppState,
      flow: {
        ...initialAppState.flow,
        memberId: '123',
        userHasAccount: true,
      },
    };

    afterEach(() => {
      jest.clearAllMocks();
    });

    it('should call zip query if memberId exists', async () => {
      const mockAppDispatch = jest.fn();
      const spy = jest.spyOn(AppStateComponent, 'useAppDispatch').mockReturnValue(mockAppDispatch);
      const { renderFn } = preRenderPage({
        appState: {
          ...initialAppState,
          flow: {
            ...initialAppState.flow,
            memberId: '123',
          },
        },
        mocks: [getZipCodeMock],
      });

      renderFn(WhoNeedsInfacilityCare);

      await waitFor(() =>
        expect(mockAppDispatch).toHaveBeenCalledWith({
          type: 'setZipcode',
          zipcode: '90001',
        })
      );

      spy.mockRestore();
    });

    it('should go to urgency screen', async () => {
      const { renderFn, routerMock } = preRenderPage({
        appState: nthDayState,
        mocks: [getSeekerInfoMock, getZipCodeMock],
      });

      renderFn(WhoNeedsInfacilityCare);

      expect(await screen.findByText('Who needs care?')).toBeInTheDocument();

      fireEvent.click(screen.getByText('Other'));
      expect(screen.getByRole('button', { name: 'Next' })).toBeEnabled();
      fireEvent.click(screen.getByRole('button', { name: 'Next' }));

      expect(routerMock?.push).toHaveBeenCalledTimes(1);
      expect(routerMock?.push).toHaveBeenLastCalledWith(SEEKER_IN_FACILITY_ROUTES.URGENCY);
    });

    it('should fire screen view event mhp_module', async () => {
      jest.spyOn(AnalyticsHelper, 'logEvent');
      const { renderFn } = preRenderPage({
        appState: nthDayState,
      });

      renderFn(WhoNeedsInfacilityCare);
      expect(AnalyticsHelper.logEvent).toHaveBeenCalledWith({
        name: 'Screen Viewed',
        data: {
          lead_flow: 'mhp module',
        },
      });
    });
  });
});
