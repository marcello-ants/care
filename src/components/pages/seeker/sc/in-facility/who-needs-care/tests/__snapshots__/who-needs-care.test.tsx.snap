// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Who needs care In facility matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root makeStyles-letsGetStartedHeader-3 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2
        class="MuiTypography-root makeStyles-header-6 MuiTypography-h2"
      >
        Letâ€™s get started by learning more about your senior care needs
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Who needs care?
      </h4>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <ul
        class="MuiList-root makeStyles-selector-1 MuiList-padding"
      >
        <li
          class="MuiListItem-root makeStyles-root-8 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-11"
            id="selector.careType.0"
            name="careType"
            readonly=""
            type="radio"
            value="PARENT"
          />
          <div
            class="makeStyles-controlIndicator-12 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  My parent
                </p>
              </div>
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-8 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-11"
            id="selector.careType.1"
            name="careType"
            readonly=""
            type="radio"
            value="SPOUSE"
          />
          <div
            class="makeStyles-controlIndicator-12 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  My spouse
                </p>
              </div>
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-8 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-11"
            id="selector.careType.2"
            name="careType"
            readonly=""
            type="radio"
            value="SELF"
          />
          <div
            class="makeStyles-controlIndicator-12 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Myself
                </p>
              </div>
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-8 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-11"
            id="selector.careType.3"
            name="careType"
            readonly=""
            type="radio"
            value="OTHER"
          />
          <div
            class="makeStyles-controlIndicator-12 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Other
                </p>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="MuiGrid-root makeStyles-ageHeader-4 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        How old are they?
      </h4>
    </div>
    <div
      class="MuiGrid-root makeStyles-stepper-2 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-ageSelector-5"
      >
        <div
          aria-haspopup="listbox"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          role="button"
          tabindex="0"
        >
          80's
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          tabindex="-1"
          value="EIGHTIES"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="makeStyles-buttonContainer-33 makeStyles-buttonContainer-34"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge Mui-disabled MuiButton-fullWidth Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Next
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
